test/python/compiler/test_disassembler.py

```diff
@@ -231,6 +243,157 @@ def test_multiple_conditionals_multiple_registers(self):
         self.assertEqual(circuits[0], qc)
         self.assertEqual({}, header)
 
+    def assertCircuitCalibrationsEqual(self, circuits, output_circuits):
+        """Verify circuit calibrations are equivalent pre-assembly and post-disassembly"""
+        self.assertTrue(
+            all(
+                len(circuit.calibrations) == len(out_qc.calibrations)
+                for circuit, out_qc in zip(circuits, output_circuits)
+            ),
+        )
+        self.assertTrue(
+            all(
+                circuit.calibrations.keys() == out_qc.calibrations.keys()
+                for circuit, out_qc in zip(circuits, output_circuits)
+            ),
+        )
+        self.assertEqual(
+            {tuple(qc_cal.keys()) for qc in circuits for qc_cal in qc.calibrations.values()},
+            {
+                tuple(out_qc_cal.keys())
+                for out_qc in output_circuits
+                for out_qc_cal in out_qc.calibrations.values()
+            },
+        )
```

Comment:
```suggestion
        for circuit, out_qc in zip(circuits, output_circuits):
            self.assertEqual(circuit.calibrations.keys(), out_qc.calibrations.keys())
            for gate_name, gate_cals in circuit.calibrations.items():
                self.assertEqual(gate_cals.keys(), out_qc.calibrations[gate_name].keys())
```
I like list comprehensions, but often it can be harder to read that way, making it harder to catch bugs.

I think the way you had written it before might return True even when the calibrations don't match, for instance:
```
# circuit.calibrations
{'rx': {((0,), ()): Schedule(...)},
  'ry': {((1,), ()): Schedule(...)}}
# out_qc.calibrations
{'rx': {((1,), ()): Schedule(...)},
  'ry': {((0,), ()): Schedule(...)}}
```
The keys are the same, `'rx'` and `'ry'`, and the nested keys are the same (`((0,), ())` and `((1,), ())`), but they don't match up (in `circuit` RX is calibrated for Q0 and in `out_qc` RX is calibrated for Q1)

Reply:
This comment ^^^ Had gotten lost in one of the rounds of code review, when an edit to the same lines marked it as "outdated"